common/document_parser/process.py

```diff
@@ -178,7 +185,61 @@ def process_dir(
             pool = multiprocessing.Pool(
                 processes=int(multiprocess), maxtasksperchild=1)
         doc_logger.info("Processing pool: %s", str(pool))
-        pool.map(single_process, data_inputs, 5)
+        
+        # ReOCR PDF if need (ex: page is missing)
+        start_ocr_time = datetime.now()
+        start_ocr_time_display = start_ocr_time.strftime("%H:%M:%S")
+        print("Start reOCR Time =", start_ocr_time_display)
+
+
+        # How many elements each list should have # work around with issue on queue being over filled
+        # using list comprehension
+        process_list = [data_inputs[i * batch_size:(i + 1) * batch_size] for i in range((len(data_inputs) + batch_size - 1) // batch_size)]
+        
+        total_num_files = len(data_inputs)
+        reocr_count = 0
+        for item_process in tqdm(process_list):
+            with concurrent.futures.ProcessPoolExecutor(max_workers=5) as executor:
+                results = [executor.submit(check_ocr_status_job_type, data[1]) for data in item_process]
+                index = 0
+                for fut in concurrent.futures.as_completed(results):
```

Comment:
might want to do `for index, fut in concurrent.futures.as_completed(results): ` so you don't have to do the index+=1 iteration

Reply:
Good thought. Made that change.